### PR TITLE
- [embree3] fix clang-cl by converting resource to UTF-8

### PR DESCRIPTION
**Pull request by SSE4**
closes: #9808

Specify library name and version:  **embree3/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/embree3/all/conanfile.py [hunk 1]
```diff
@@ -170,6 +170,13 @@ def _configure_cmake(self):
         return self._cmake
 
     def build(self):
+        # some compilers (e.g. clang) do not like UTF-16 sources
```
Comment (SSE4) [inline recipes/embree3/all/conanfile.py:173]: for an additional safety, we may put it under `if self.settings.compiler == "clang'`. but I've checked, and Visual Studio builds just fine with it.

File: recipes/embree3/all/conanfile.py [hunk 2]
```diff
@@ -170,6 +171,13 @@ def _configure_cmake(self):
...
+        rc = os.path.join(self._source_subfolder, "kernels", "embree.rc")
+        content = load(self, rc, encoding="utf_16_le")
+        if content[0] == '\ufeff':
+            content = content[1:]
+        content = "#pragma code_page(65001)\n" + content
+        save(self, rc, content, encoding="utf8")
```
Comment (SSE4) [inline recipes/embree3/all/conanfile.py:180]:
```suggestion
        save(self, rc, content)
```

**Review by uilianries — APPROVED** [review]
Tricky

**Review by AndreyMlashkin — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@SpaceIm @danimtb @jgsogo 
Please approve

**Review by jgsogo — APPROVED** [review]
this change should probably be reported upstream

**Review by intelligide — APPROVED** [review]

**Pull request merged**